### PR TITLE
Add Rustls support

### PR DESCRIPTION
**Pull request by KeenS**
fix ~~#812~~ #128 . This patch bumps MSRV to 1.56.0 because rustls requires it.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,11 +63,21 @@ jobs:
...
       - run:
-          name: Unit testing
+          name: Unit testing with ssl feature
           command: |
             RUST_BACKTRACE=1 cargo test -j 8 --verbose --features ssl
       - run:
-          name: More unit testing
+          name: Unit testing with ssl-rustls feature
+          command: |
+            RUST_BACKTRACE=1 cargo test -j 8 --verbose --features ssl
```
Comment (eldesh) [inline .circleci/config.yml:72]: Is your intention to specify the ssl-rustls option?

File: .circleci/config.yml [hunk 2]
```diff
@@ -63,11 +63,21 @@ jobs:
...
+          name: More unit testing with ssl feature
+          command: |
+            docker load -i test-iostream
+            docker load -i test-signal
+            RUST_BACKTRACE=1 cargo test --verbose --features ssl -- --ignored
+      - run:
+          name: More unit testing with ssl-rustls feature
           command: |
             docker load -i test-iostream
             docker load -i test-signal
```
Comment (eldesh) [inline .circleci/config.yml:83]: Save above (ssl-rustls?).
And in this point, test-iostrem and test-signal are loaded already at `More unit testing with ssl feature`.

**Comment by eldesh** [comment]
(Are you talking about the #128 ?)

File: .circleci/config.yml (hunk 1 above)
Comment (KeenS) [reply .circleci/config.yml:72]: lol

**Pull request closed without merging**